ui/src/StateContainer.js

```diff
@@ -67,6 +72,34 @@ export class StateContainer extends LitElement {
         ).shadowRoot;
     }
 
+    /**
+     * Parses the URL for relevant project-related details. 
+     */
+    setProjectDetails() {
+        this.projectId = this.urlService.getProjectId();
+        this.userLang = this.urlService.getLang();
+        this.vendorId = this.urlService.getVendorId();
+    }
+
+    /**
+     * If present and different from that stored in the cookies, give
+     * the userId provided in the URL priority and clear every other
+     * component of user information. 
+     */
+    setUserId() {
```

Comment:
nit: setUserInfoOntoCookie? Since we're not just dealing with userId here.

Reply:
good point--adjusted